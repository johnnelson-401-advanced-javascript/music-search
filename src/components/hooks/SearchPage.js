import React from 'react';
import ArtistList from '../Artist/ArtistList';
import Form from '../Form/Form';
import useArtists from '../custom-hooks/useArtists';
import usePaging from '../custom-hooks/usePaging';

export default function SearchPage() {

  const { page, handlePageBackward, handlePageForward } = usePaging();
  const { artists, loading, searchQuery, handleChange, handleSubmit } = useArtists(page);
  
  if(loading) return <img src='https://loading.io/spinners/music/lg.music-note-preloader.gif' />;

  return (
    <>
      <Form
        handleChange={handleChange}
        handleSubmit={handleSubmit}
        searchQuery={searchQuery} />
      <ArtistList
        handlePageBackward={handlePageBackward}
        handlePageForward={handlePageForward}
        artists={artists}
      />
    </>
  );
}
