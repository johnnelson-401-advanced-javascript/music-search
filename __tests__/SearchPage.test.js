import React from 'react';
import { shallow } from 'enzyme';
import SearchPage from '../src/components/hooks/SearchPage';

jest.mock('../src/services/musicBrainzApi.js', () => ({
  fetchArtists() {
    return Promise.resolve([
      { 
        disamb: 'test',
        name: 'Test', 
        id: '5' 
      }
    ]);
  }
}));

describe('SearchPage', () => {
  it.only('updates state with an array of artists', () => {
    const wrapper = shallow(<SearchPage />);
    expect(wrapper).toMatchSnapshot();

  });
});
