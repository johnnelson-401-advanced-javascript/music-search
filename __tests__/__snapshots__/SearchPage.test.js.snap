// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage updates state with an array of artists 1`] = `
<Fragment>
  <SearchForm
    handleChange={[Function]}
    handleSubmit={[Function]}
    searchQuery=""
  />
  <ArtistList
    artists={Array []}
    handlePageBackward={[Function]}
    handlePageForward={[Function]}
  />
</Fragment>
`;
